test/integration/createProvider_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
```

Comment:
Is there an advantage to having a distinct integration test package? Could we instead keep the integration tests closer to the controllers they're testing, and perhaps [gate them behind a build tag](https://filipnikolovski.com/separating-tests-in-go/)?

Reply:
I like this idea! However, I do think we will likely have many of our integration tests span multiple controllers, so we may want a general location as well. For instance, if we wanted to test deploying Wordpress on GCP or something on the more complex side like that (not intending to do so in initial implementation) it would be useful to have something like an `e2e` directory.